demos/rag_agentic/notebooks/extras/mcp_toolgroup_management.ipynb

```diff
@@ -0,0 +1,365 @@
+    "\n",
+    "###  What is MCP?\n",
+    "\n",
+    "Model Context Protocol (MCP) is a mechanism that allows agents to interact with tools from a registered endpoint. These MCP tools are used to extend the capabilities of an agent in real time.\n",
+    "\n",
+    "To use MCP tools, you need:\n",
+    "- A running MCP server with a valid endpoint URL.\n",
+    "- An application hosting the MCP client."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f67c1cd-9806-49ec-88f5-16e39dc9030f",
+   "metadata": {},
+   "source": [
+    "## Prerequisites "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48a0f52c-0381-4a51-831c-67fa6be98f3a",
+   "metadata": {},
+   "source": [
+    "Before starting, ensure:\n",
+    "\n",
+    "- You have a running instance of a LlamaStack server ([local](https://github.com/opendatahub-io/llama-stack-demos/blob/main/local_setup_guide.md) or [remote](https://github.com/opendatahub-io/llama-stack-demos/tree/main/kubernetes)).\n",
+    "- You have configured with the following variables:\n",
+    "```env\n",
+    "REMOTE_BASE_URL=http://your-llamastack-url\n",
+    "MCP_SERVER_URL=http://your-mcp-url\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "4401aba8-d618-4b15-abfc-25719e499e3f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from llama_stack_client import LlamaStackClient\n",
+    "from dotenv import load_dotenv\n",
+    "load_dotenv()\n",
+    "\n",
+    "base_url=os.getenv(\"REMOTE_BASE_URL\")\n",
+    "# connecting to remote server\n",
+    "client = LlamaStackClient(base_url=base_url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2391984d-4e89-4885-80d3-03d7ce7d76ab",
+   "metadata": {},
+   "source": [
+    "## View Registered tool groups"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "374cf724-695f-4340-a813-02f1466b8925",
+   "metadata": {},
+   "source": [
+    "Refer to this [kubernetes/mcp-servers/](https://github.com/opendatahub-io/llama-stack-demos/tree/main/kubernetes/mcp-servers) to see how to deploy a number of MCP servers on OpenShift.\n",
+    "\n",
+    "To inspect which toolgroups are currently registered with your LlamaStack server, use:\n",
+    "\n",
+    "```python\n",
+    "client.tools.list()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "328972fd-7338-4555-9be1-b4ee73fe19d4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Registered Toolgroups:\n",
+      " - builtin::code_interpreter\n",
+      " - mcp::github\n",
+      " - mcp::openshift\n",
+      " - builtin::websearch\n",
+      " - builtin::rag\n",
+      " - mcp::slack\n",
+      " - mcp::custom_tool\n",
+      " - mcp::ansible\n"
+     ]
+    }
+   ],
+   "source": [
+    "toolgroups = client.tools.list()\n",
+    "registered_toolgroups = set(t.toolgroup_id for t in toolgroups)\n",
```

Comment:
if you use `client.toolgroups.list()` you don't need this set comprehension in line 107